tests/functional/testSubscription.js

```diff
@@ -26,6 +27,82 @@ const SUBSCRIPTION_DETAILS = {
   },
 };
 
+describe('Subscription manager', function() {
+  describe('Expired subscription', function() {
+    this.timeout(30000);
+
+    const device = {
+      name: 'Current device',
+      unique_id: '',
+      pubkey: '',
+      ipv4_address: '127.0.0.1',
+      ipv6_address: '::1',
+      created_at: new Date().toISOString()
+    };
+
+    const userDataActive = {
+      avatar: '',
+      display_name: 'Test',
+      email: 'test@mozilla.com',
+      max_devices: 5,
+      subscriptions: {vpn: {active: true}},
+      devices: [device],
+    };
+    const userDataInactive = {
+      avatar: '',
+      display_name: 'Test test',
+      email: 'test@mozilla.com',
+      max_devices: 5,
+      subscriptions: {vpn: {active: false}},
+      devices: [device]
+    };
+
+    this.ctx.guardianOverrideEndpoints = {
+      GETs: {
+        '/api/v1/vpn/account': {
+          status: 200,
+          requiredHeaders: ['Authorization'],
+          body: userDataActive
+        }
+      },
+      POSTs: {
+        '/api/v1/vpn/device': {
+          status: 201,
+          requiredHeaders: ['Authorization'],
+          bodyValidator: validators.guardianDevice,
+          callback: (req) => {
+            device.name = req.body.name;
+            device.pubkey = req.body.pubkey;
+            device.unique_id = req.body.unique_id;
+          },
+          body: {}
+        },
+      }
+    };
+
+    it('Verify subscription before enabling VPN', async () => {
+      // This test verifies the case where a user is logged in
+      // but the VPN is off when their subscription expires.
+      // When they try to turn the VPN on, they get the
+      // "Subscribe to Mozilla VPN" screen.
+
+      // Step 1: User logs into the client successfully.
+      await vpn.authenticateInApp(true, true);
+
+      // Step 2: Override the Guardian endpoint to mock an expired subscription.
+      this.ctx.guardianOverrideEndpoints.GETs['/api/v1/vpn/account'].body =
+          userDataInactive;
+
+      // Step 3: Attempt to toggle the VPN on.
```

Comment:
Let's use the vpn.activate() helper method unless you had a reason for not.

Reply:
`vpn.activate()` implementation explicitly checks that the VPN was enabled successfully (which doesn't/shouldn't happen in this case) so it ends up error-ing out if I use it here. Instead I need to explicitly attempt to turn the VPN on, then check that we land on the "Subscription required" screen.